zerver/lib/actions.py

```diff
@@ -2569,6 +2576,22 @@ def gather_subscriptions_helper(user_profile):
         else:
             unsubscribed.append(stream_dict)
 
+    never_subscribed_stream_ids = [item for item in all_streams_id if item not in sub_unsub_stream_ids]
+    never_subscribed_streams = all_active_streams.select_related(
+        "realm").filter(id__in=never_subscribed_stream_ids)
+    for stream in never_subscribed_streams:
+        if stream.invite_only:
+            continue
```

Comment:
It'd be cleaner to filter for invite_only streams by adding `, invite_only=False)` to your filter query above.